### PR TITLE
Add nodeTemplate to MachineClass CRD

### PR DESCRIPTION
**Pull request by himanshu-kun**
**How to categorize this PR?**

/area control-plane
/kind bug

**What this PR does / why we need it**:
Adds nodeTemplate field to machineClass CRD . Without this the scale-from-zero feature won't work
**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```bugfix dependency
MachineClass CRD has been updated with `nodeTemplate` field. This is essential for scale-from-zero feature of CA to work.
```

### CODE REVIEW

**Comment by himanshu-kun** [comment]
Just confirming that this change will go as part of patch release `v1.38.3`?

File: extensions/pkg/controller/worker/templates/machineclasses.tpl.yaml [hunk 1]
```diff
@@ -54,13 +54,44 @@ spec:
                   must be unique.
                 type: string
             type: object
+            x-kubernetes-preserve-unknown-fields: true
```
Comment (rfranzke) [inline extensions/pkg/controller/worker/templates/machineclasses.tpl.yaml]: Why was this added here to `credentialsSecretRef`? 👀

File: extensions/pkg/controller/worker/templates/machineclasses.tpl.yaml [hunk 2]
```diff
@@ -54,13 +54,44 @@ spec:
...
+              zone:
+                description: Zone of the expected node belonging to nodeGroup
+                type: string
+            required:
+            - capacity
+            - instanceType
+            - region
+            - zone
+            type: object
+            x-kubernetes-preserve-unknown-fields: true
```
Comment (rfranzke) [inline extensions/pkg/controller/worker/templates/machineclasses.tpl.yaml:93]: Why is this line needed?

Comment (himanshu-kun) [reply extensions/pkg/controller/worker/templates/machineclasses.tpl.yaml:93]: Without `x-kubernetes-preserve-unknown-fields: true` the contents of `nodeTemplate` will get truncated. Please refer [this PR](https://github.com/gardener/machine-controller-manager/pull/647) for details.

File: extensions/pkg/controller/worker/templates/machineclasses.tpl.yaml (hunk 1 above)
Comment (himanshu-kun) [reply extensions/pkg/controller/worker/templates/machineclasses.tpl.yaml]: This was also part of the mentioned PR , but I think things are working fine without it as well. Shall I remove it?

Comment (rfranzke) [reply extensions/pkg/controller/worker/templates/machineclasses.tpl.yaml]: I've read the PR and it is explicitly mentioned there:

> Also, previously, the credentialSecretRef didn't have this tag x-kubernetes-preserve-unknown-fields too. Added that as well.

I'm confused why this was added in the first place, but now you say it can also be removed. What now? If it is not needed (which I think it is) then let's remove it. I suggest to then also remove it from g/mcm.

File: extensions/pkg/controller/worker/templates/machineclasses.tpl.yaml (hunk 2 above)
Comment (rfranzke) [reply extensions/pkg/controller/worker/templates/machineclasses.tpl.yaml:93]: Just to confirm: Is this because `capacity` is of type `corev1.ResourceList` which itself is just a `map` and can contain arbitrary keys?

File: extensions/pkg/controller/worker/templates/machineclasses.tpl.yaml (hunk 1 above)
Comment (himanshu-kun) [reply extensions/pkg/controller/worker/templates/machineclasses.tpl.yaml]: Yes , I also didn't see any benefit for adding it at that time. Sure I will remove it from g/mcm as well.

File: extensions/pkg/controller/worker/templates/machineclasses.tpl.yaml (hunk 2 above)
Comment (himanshu-kun) [reply extensions/pkg/controller/worker/templates/machineclasses.tpl.yaml:93]: yes you are correct.

Comment (rfranzke) [reply extensions/pkg/controller/worker/templates/machineclasses.tpl.yaml:93]: OK, thanks!

File: extensions/pkg/controller/worker/templates/machineclasses.tpl.yaml (hunk 1 above)
Comment (rfranzke) [reply extensions/pkg/controller/worker/templates/machineclasses.tpl.yaml]: OK, thanks!

**Review by rfranzke — APPROVED** [review]
/lgtm

**Review by ialidzhikov — APPROVED** [review]
/lgtm
